File: sort/SelectSort/selectsort.js
const selectSort = (arr) => {
    for (let i = 0; i < arr.length - 1; i++) {
        let minIndex = i;
        for (let j = i + 1; j < arr.length; j++) {
            if (arr[j] < arr[minIndex]) {
                minIndex = j;
            }
        }
        if (minIndex !== i) {
            [arr[i], arr[minIndex]] = [arr[minIndex], arr[i]]
        }
    }
    console.log(arr)
}

(function () {
    const arr = [4, 2, 3, 1, 6, 7, 9];
    selectSort(arr);
})();

File: sort/SelectSort/selectsort.go
package main

import "fmt"

func selectSort(arr []int) {
	for i := 0; i < len(arr); i++ {
		minIndex := i

		for j := i; j < len(arr); j++ {
			if arr[j] < arr[minIndex] {
				minIndex = j
			}
		}

		if minIndex != i {
			arr[minIndex], arr[i] = arr[i], arr[minIndex]
		}
	}

	fmt.Println(arr)
}

func main() {
	arr := []int{4, 2, 3, 1, 6, 7, 9}
	selectSort(arr)
}
